shared/src/pool_fetching.rs

```diff
@@ -20,30 +21,38 @@ type RelativeReserves = (U256, U256, H160);
 
 #[async_trait::async_trait]
 pub trait PoolFetching: Send + Sync {
-    async fn fetch(&self, token_pairs: HashSet<TokenPair>) -> Vec<Pool>;
+    async fn fetch(&self, token_pairs: HashSet<TokenPair>) -> Result<Vec<Pool>>;
```

Comment:
This change makes the use of this component more complicated. Do we actually still need the BadTokenDetector here if we have it in the other components?

Reply:
One possible solution I see would be to make the `BadTokenDetecting` infallible. The implementation with `trace_callMany` could then give the token the benefit of the doubt on unrelated failures (connection issues for example).

Additionally, I would also be curious, as @fleupold asked, if we need `PoolFetching` to check the quality of the tokens. Wouldn't it make sense for the caller to be required only fetch pools for "good" tokens (i.e. the filtering happening "higher-up"). I understand that the pool fetcher previously filtered bad tokens, so removing it would be unrelated to this change. It could go in a follow up PR?

That being said, I'm OK with returning a `Result` here.

Reply:
Fixed.